xrpl/utils/txn_parser/utils/balance_parser.py

```diff
@@ -0,0 +1,161 @@
+"""Helper functions for balance parser."""
+
+from decimal import Decimal
+from typing import Callable, List, Optional
+
+from pydash import group_by  # type: ignore
+
+from xrpl.utils.txn_parser.utils.nodes import NormalizedNode
+from xrpl.utils.txn_parser.utils.types import Balance, BalanceChange, BalanceChanges
+from xrpl.utils.xrp_conversions import drops_to_xrp
+
+
+def _get_xrp_quantity(
+    node: NormalizedNode,
+    value_parser: Callable[[NormalizedNode], Optional[Decimal]],
+) -> Optional[BalanceChange]:
+    value = value_parser(node)
+    if value is None:
+        return None
+    value_is_negative = value < Decimal(0)
```

Comment:
(This)[https://docs.python.org/3/library/decimal.html#decimal.Decimal.is_signed] might be a better way of doing that